apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php

```diff
@@ -148,4 +161,18 @@ private function get_block_patterns_locale() {
 		$language = function_exists( 'get_blog_lang_code' ) ? get_blog_lang_code() : get_locale();
 		return \A8C\FSE\Common\get_iso_639_locale( $language );
 	}
+
+	/**
+	 * Check that the pattern is allowed to be registered.
+	 *
+	 * @param array $pattern    A pattern with a 'tags' array where the key is the tag slug in English.
+	 *
+	 * @return bool
+	 */
+	private function can_register_pattern( $pattern ) {
+		if ( $this->skip_full_width_patterns && in_array( 'requires_align_wide', array_keys( $pattern['tags'] ), true ) ) {
```

Comment:
Question: if we're expecting a _specific_ tag anyway, might a future optimization be (assuming they'll be more of such theme exceptions) something like `requires_{ $feature }`, and then

```
	private function can_register_pattern( $pattern ) {
        foreach( $pattern['tags'] as $tag ) {
            $exploded_tag = explode( 'requires_', $tag );
            if ( $exploded_tag[1] ) {
                return false !== get_theme_support( $exploded_tag[1] )
            }
        }
		return true;
	}
```

Very willing to be accused of over-thinking things! :D

Reply:
@ramonjd I actually quite like that. I'll have a play with it now, because it seems fairly safe and will also future proof it for any similar situations (the issue can be resolved by adding / removing tags, rather than requiring further code changes). Initially I thought it'd be better to cache the `get_theme_support` feature rather than call it on every pattern (since we'll have > 100 patterns), but from looking at how `get_theme_support` [is implemented](https://developer.wordpress.org/reference/functions/get_theme_support/), it's an associative array lookup rather than looping over anything, so seems like it shouldn't be too expensive.

It just means we'd use the tag `requires-align-wide` instead of `requires_align_wide` so that we're hyphenating correctly.

Reply:
> Initially I thought it'd be better to cache the get_theme_support feature rather than call it on every pattern (since we'll have > 100 patterns),


Ah yeah, good point. That totally escaped me. 

Anyway, it's just an idea seeing as we're expecting a specific string regardless. 

Sorry if I threw an implementation pepper into the soup!

Reply:
No, no, it's a good idea, thank you! Better to spend a little bit longer getting it right now, so that we can save some time further down the track 😀